File: app/src/main/java/com/summerland/android/game/BackGround.java
package com.summerland.android.game;

import android.graphics.Bitmap;
import android.graphics.Canvas;

/**
 * Created by steve on 01/10/16.
 */
public class BackGround {

    public static final int WIDTH = 856;
    public static final int HEIGHT = 480;

    private Bitmap image;
    private int x, y, dx;

    public BackGround(Bitmap res){
        image = res;
    }

    public BackGround(Bitmap res, int vector) {
        image = res;
        dx = vector;
    }

    public void update(){
        x += dx;
        if (x < -BackGround.WIDTH) x = 0;
    }

    public void draw(Canvas canvas){
        canvas.drawBitmap(image, x, y, null);
        if (x < 0) canvas.drawBitmap(image, x+BackGround.WIDTH, y, null);
    }

    public void setVector(int dx){
        this.dx = dx;
    }
}

File: app/src/main/java/com/summerland/android/game/BackGround.kt
package com.summerland.android.game

import android.graphics.Bitmap
import android.graphics.Canvas

/**
 * BackGround class, Created by steve on 01/10/16.
 */
class BackGround internal constructor(private val image: Bitmap, private val dx: Int) {
    private var x: Int = 0
    private val y: Int = 0

    fun update() {
        x += dx
        if (x < -BackGround.WIDTH) x = 0
    }

    fun draw(canvas: Canvas) {
        canvas.drawBitmap(image, x.toFloat(), y.toFloat(), null)
        if (x < 0) canvas.drawBitmap(image, (x + BackGround.WIDTH).toFloat(), y.toFloat(), null)
    }

    companion object {

        internal val WIDTH = 856
        internal val HEIGHT = 480
    }

    //public void setVector(int dx){ this.dx = dx; }
}

